**Karl Darwin Ombele Ndassebe**

**MELHORANDO A EFICÁCIA DA CADEIA DE SUPRIMENTOS: Um Estudo de Caso em Moda e Beleza**

*03 de setembro, 2024*

**1. INTRODUÇÃO**

Neste estudo, realizei uma análise detalhada da cadeia de suprimentos para uma startup inovadora no setor de Moda e Beleza. O objetivo principal foi examinar o conjunto de dados fornecido pela empresa, com o intuito de identificar tendências, extrair insights relevantes e propor melhorias que aumentem a eficácia da cadeia de suprimentos. Ao otimizar processos e tomar decisões baseadas em dados, busquei criar mais valor para os clientes, alinhando as operações da cadeia de suprimentos com as exigências de um mercado dinâmico e competitivo. Esta análise é fundamental para apoiar a estratégia da empresa em sua missão de entregar produtos de alta qualidade com eficiência e consistência, mantendo-se ágil diante das mudanças nas preferências dos consumidores e nas condições do mercado.

**2. CARREGAMENTO E TRATAMENTO DOS DADOS**

**2.1 Importação das bibliotecas necessárias**

Inicei o projeto com a importação de todas as bibliotecas necessárias para a realização da análise.

In [220]:
# Importando as bibliotecas necessárias

import pandas as pd
import numpy as np
import plotly.express as px
import plotly.io as pio
import plotly.graph_objects as go 

# Configurando o tema padrão dos gráficos do Plotly

pio.templates.default = "plotly_white"

**2.2 Carrregamento do conjunto de dados**

In [221]:
TabcadeiaSup = pd.read_excel("Cadeia de Suprimentos.xlsx")

**2.3 Inspeção do conjunto de dados**

In [222]:
# Exibindo o dataset para verificação inicial

print("Visualização inicial dos dados:")
display(TabcadeiaSup)

Visualização inicial dos dados:


Tipo de produto Código do produto (SKU)    Preço  Disponibilidade  \
0   Cuidados com o cabelo                    SKU0  69.8080               55   
1     Cuidados com a pele                    SKU1  14.8435               95   
2   Cuidados com o cabelo                    SKU2  11.3197               34   
3     Cuidados com a pele                    SKU3  61.1633               68   
4     Cuidados com a pele                    SKU4   4.8055               26   
..                    ...                     ...      ...              ...   
95  Cuidados com o cabelo                   SKU95  77.9039               65   
96             Cosméticos                   SKU96  24.4231               29   
97  Cuidados com o cabelo                   SKU97   3.5261               56   
98    Cuidados com a pele                   SKU98  19.7546               43   
99  Cuidados com o cabelo                   SKU99  68.5178               17   

    N° de produtos vendidos  Receita gerada Demografia dos clientes  \
0                       802       8661.9968             Não-binário   
1                       736       7460.9001                Feminino   
2                         8       9577.7496            Desconhecido   
3                        83       7766.8364             Não-binário   
4                       871       2686.5052             Não-binário   
..                      ...             ...                     ...   
95                      672       7386.3639            Desconhecido   
96                      324       7698.4248             Não-binário   
97                       62       4370.9166               Masculino   
98                      913       8525.9526                Feminino   
99                      627       9185.1858            Desconhecido   

    Níveis de estoque  Prazos de entrega  Quantidades de pedido  ...  \
0                  58                  7                     96  ...   
1                  53                 30                     37  ...   
2                   1                 10                     88  ...   
3                  23                 13                     59  ...   
4                   5                  3                     56  ...   
..                ...                ...                    ...  ...   
95                 15                 14                     26  ...   
96                 67                  2                     32  ...   
97                 46                 19                      4  ...   
98                 53                  1                     27  ...   
99                 55                  8                     59  ...   

       Localização Prazo de entrega  Volumes de produção Prazo de produção  \
0        São Paulo               29                  215                29   
1        São Paulo               23                  517                30   
2        São Paulo               12                  971                27   
3   Rio de Janeiro               24                  937                18   
4         Brasília                5                  414                 3   
..             ...              ...                  ...               ...   
95       São Paulo               18                  450                26   
96       São Paulo               28                  648                28   
97       São Paulo               10                  535                13   
98        Salvador               28                  581                 9   
99        Salvador               29                  921                 2   

   Custos de fabricação  Resultados da inspeção  Taxas de defeitos  \
0               46.2799                Pendente           0.226410   
1               33.6168                Pendente           4.854068   
2               30.6880                Pendente           4.580593   
3               35.6247              Reprovado            4.746649   
4               92.0652              Reprovado          

In [223]:
# Verificando as estatísticas descritivas iniciais

print("\nEstatísticas descritivas do conjunto de dados:")
TabcadeiaSup.describe()


Estatísticas descritivas do conjunto de dados:


Preço  Disponibilidade  N° de produtos vendidos  Receita gerada  \
count  100.000000       100.000000               100.000000      100.000000   
mean    49.462456        48.400000               460.990000     5776.048189   
std     31.168190        30.743317               303.780074     2732.841744   
min      1.700000         1.000000                 8.000000     1061.618500   
25%     19.597825        22.750000               184.250000     2812.847125   
50%     51.239850        43.500000               392.500000     6006.352050   
75%     77.198225        75.000000               704.250000     8253.976900   
max     99.171300       100.000000               996.000000     9866.465500   

       Níveis de estoque  Prazos de entrega  Quantidades de pedido  \
count         100.000000         100.000000             100.000000   
mean           47.770000          15.960000              49.220000   
std            31.369372           8.785801              26.784429   
min             0.000000           1.000000               1.000000   
25%            16.750000           8.000000              26.000000   
50%            47.500000          17.000000              52.000000   
75%            73.000000          24.000000              71.250000   
max           100.000000          30.000000              96.000000   

       Tempos de envio  Custos de envio  Prazo de entrega  \
count       100.000000       100.000000        100.000000   
mean          5.750000         5.548146         17.080000   
std           2.724283         2.651369          8.846251   
min           1.000000         1.013500          1.000000   
25%           3.750000         3.540225         10.000000   
50%           6.000000         5.320550         18.000000   
75%           8.000000         7.601650         25.000000   
max          10.000000         9.929800         30.000000   

       Volumes de produção  Prazo de produção  Custos de fabricação  \
count           100.000000          100.00000            100.000000   
mean            567.840000           14.77000             47.266698   
std             263.046861            8.91243             28.982847   
min             104.000000            1.00000              1.085100   
25%             352.000000            7.00000             22.983325   
50%             568.500000           14.00000             45.905650   
75%             797.000000           23.00000             68.621000   
max             985.000000           30.00000             99.466100   

       Taxas de defeitos      Custos  
count         100.000000  100.000000  
mean            2.277158  529.245782  
std             1.461366  258.301701  
min             0.018608  103.916200  
25%             1.009650  318.778450  
50%             2.141863  520.430450  
75%             3.563995  763.078250  
max             4.939255  997.413500

In [224]:
# Verificando a estrutura do conjunto de dados

print("Estrutura do conjunto de dados:")
TabcadeiaSup.info()

Estrutura do conjunto de dados:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 24 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Tipo de produto          100 non-null    object 
 1   Código do produto (SKU)  100 non-null    object 
 2   Preço                    100 non-null    float64
 3   Disponibilidade          100 non-null    int64  
 4   N° de produtos vendidos  100 non-null    int64  
 5   Receita gerada           100 non-null    float64
 6   Demografia dos clientes  100 non-null    object 
 7   Níveis de estoque        100 non-null    int64  
 8   Prazos de entrega        100 non-null    int64  
 9   Quantidades de pedido    100 non-null    int64  
 10  Tempos de envio          100 non-null    int64  
 11  Transportadoras          100 non-null    object 
 12  Custos de envio          100 non-null    float64
 13  Nome do fornecedor       100 non-null    object 


O conjunto de dados contém 100 entradas e 24 colunas, abrangendo uma ampla gama de informações relacionadas à cadeia de suprimentos. Cada coluna possui 100 valores não nulos, indicando que não há dados ausentes. As colunas incluem variáveis categóricas, como "Tipo de produto" e "Nome do fornecedor", bem como variáveis numéricas, como "Preço" e "Custos de fabricação". A estrutura do DataFrame é composta por 6 colunas do tipo float64, 9 colunas do tipo int64, e 9 colunas do tipo object. 

In [225]:
# Confirmando ausência de valores ausentes

print("\nVerificçao de valores ausentes:")
print(TabcadeiaSup.isnull().sum())


Verificçao de valores ausentes:
Tipo de produto            0
Código do produto (SKU)    0
Preço                      0
Disponibilidade            0
N° de produtos vendidos    0
Receita gerada             0
Demografia dos clientes    0
Níveis de estoque          0
Prazos de entrega          0
Quantidades de pedido      0
Tempos de envio            0
Transportadoras            0
Custos de envio            0
Nome do fornecedor         0
Localização                0
Prazo de entrega           0
Volumes de produção        0
Prazo de produção          0
Custos de fabricação       0
Resultados da inspeção     0
Taxas de defeitos          0
Modos de transporte        0
Rotas                      0
Custos                     0
dtype: int64


In [226]:
# Verificando valores únicos em colunas categóricas

categorical_columns = [
    'Tipo de produto',
    'Código do produto (SKU)',
    'Demografia dos clientes',
    'Transportadoras',
    'Nome do fornecedor',
    'Localização',
    'Rotas'
]

for column in categorical_columns:
    print(f"\nValores únicos na coluna {column}:")
    print(TabcadeiaSup[column].unique())


Valores únicos na coluna Tipo de produto:
['Cuidados com o cabelo' 'Cuidados com a pele' 'Cosméticos']

Valores únicos na coluna Código do produto (SKU):
['SKU0' 'SKU1' 'SKU2' 'SKU3' 'SKU4' 'SKU5' 'SKU6' 'SKU7' 'SKU8' 'SKU9'
 'SKU10' 'SKU11' 'SKU12' 'SKU13' 'SKU14' 'SKU15' 'SKU16' 'SKU17' 'SKU18'
 'SKU19' 'SKU20' 'SKU21' 'SKU22' 'SKU23' 'SKU24' 'SKU25' 'SKU26' 'SKU27'
 'SKU28' 'SKU29' 'SKU30' 'SKU31' 'SKU32' 'SKU33' 'SKU34' 'SKU35' 'SKU36'
 'SKU37' 'SKU38' 'SKU39' 'SKU40' 'SKU41' 'SKU42' 'SKU43' 'SKU44' 'SKU45'
 'SKU46' 'SKU47' 'SKU48' 'SKU49' 'SKU50' 'SKU51' 'SKU52' 'SKU53' 'SKU54'
 'SKU55' 'SKU56' 'SKU57' 'SKU58' 'SKU59' 'SKU60' 'SKU61' 'SKU62' 'SKU63'
 'SKU64' 'SKU65' 'SKU66' 'SKU67' 'SKU68' 'SKU69' 'SKU70' 'SKU71' 'SKU72'
 'SKU73' 'SKU74' 'SKU75' 'SKU76' 'SKU77' 'SKU78' 'SKU79' 'SKU80' 'SKU81'
 'SKU82' 'SKU83' 'SKU84' 'SKU85' 'SKU86' 'SKU87' 'SKU88' 'SKU89' 'SKU90'
 'SKU91' 'SKU92' 'SKU93' 'SKU94' 'SKU95' 'SKU96' 'SKU97' 'SKU98' 'SKU99']

Valores únicos na coluna Demografia dos cl


O conjunto de dados contém diversas categorias: os produtos são classificados como "Cuidados com o cabelo", "Cuidados com a pele" e "Cosméticos". Cada item tem um código SKU exclusivo entre 100 opções disponíveis. A demografia dos clientes inclui "Não-binário", "Feminino", "Desconhecido" e "Masculino". Os produtos são enviados por três transportadoras distintas e vêm de vários fornecedores e localidades, como São Paulo e Rio de Janeiro. Além disso, há três rotas principais usadas para o transporte.

**3. ANÁLISE EXPLORATÓRIA DE DADOS (EDA)**

In [227]:
fig = px.scatter(TabcadeiaSup, 
                 x='Preço', 
                 y='Receita gerada', 
                 color='Tipo de produto', 
                 hover_data=['N° de produtos vendidos'], 
                 trendline="ols")
fig.show()

O gráfico revela que os produtos de "Cuidados com os cabelos" têm uma sensibilidade maior ao preço, com a receita diminuindo conforme o preço aumenta, sugerindo uma maior sensibilidade ao preço pelos consumidores. Já os produtos de "Cuidados com a pele" mostram uma tendência positiva, indicando que um aumento no preço pode não apenas ser aceito pelos consumidores, mas também potencialmente aumentar a receita, refletindo uma percepção de maior valor. Os "Cosméticos" seguem uma tendência similar aos produtos de "Cuidados com os cabelos", com uma leve redução na receita à medida que o preço sobe, embora essa diminuição seja menos acentuada.

In [228]:
# Agrupando os dados por tipo de produto e somando o número de produtos vendidos
Vendas_por_Tipo_de_produto = TabcadeiaSup.groupby('Tipo de produto')['N° de produtos vendidos'].sum().reset_index()

# Criando o gráfico de rosca
pie_chart = px.pie(Vendas_por_Tipo_de_produto, 
                   values='N° de produtos vendidos', 
                   names='Tipo de produto', 
                   title='Vendas por Tipo de produto', 
                   hole=0.5, 
                   )

# Atualizando o gráfico para exibir percentuais e rótulos dentro dos setores
pie_chart.update_traces(textposition='outside', textinfo='percent+label', 
                        marker=dict(line=dict(color='#000000', width=1)))

O gráfico revela que "Cuidados com a Pele" é a categoria dominante em vendas, respondendo por 45% do total, o que indica uma forte demanda por esses produtos. "Cuidados com o Cabelo" ocupa a segunda posição, com 29,5% das vendas, evidenciando uma demanda também significativa. Já "Cosméticos" representa a menor participação, com 25,5%, embora ainda constitua uma parte relevante das vendas totais.

In [229]:
# Agrupando os dados por Demografia dos clientes e somando as receitas geradas
Receitas_geradas_por_Genero = TabcadeiaSup.groupby('Demografia dos clientes')['Receita gerada'].sum().reset_index()

# Criando o gráfico de barras horizontal
Receitas_geradas_por_Genero_chart = px.bar(Receitas_geradas_por_Genero, 
                   x='Receita gerada', 
                   y='Demografia dos clientes', 
                   orientation='h', 
                   title='Distribuição de Receita por Gênero e Identidade dos Clientes', 
                   color='Demografia dos clientes',
                   color_discrete_sequence=['#1f77b4'])

# Atualizando o layout para melhor visualização
Receitas_geradas_por_Genero_chart.update_layout(xaxis_title='Receita Gerada', 
                        yaxis_title='Demografia dos Clientes',
                        showlegend=False)
Receitas_geradas_por_Genero_chart.show()

A análise revela que a maior parte da receita é gerada por clientes cujo gênero está registrado como "Desconhecido", destacando a importância de entender melhor esse segmento. O grupo "Feminino" é o segundo mais significativo em termos de receita, demonstrando uma forte presença de consumidoras. Em seguida, os clientes do gênero "Masculino" também têm um impacto relevante nas vendas. Por último, o grupo "Não-binário" contribui com a menor parcela da receita, possivelmente devido a uma menor representatividade ou diferentes padrões de consumo.

In [230]:
# Agrupando os dados por Transportadoras e somando as receitas geradas
Receitas_geradas_por_Transportadora = TabcadeiaSup.groupby('Transportadoras')['Receita gerada'].sum().reset_index()

# Criando o gráfico de barras
fig = go.Figure()
fig.add_trace(go.Bar(x=Receitas_geradas_por_Transportadora['Transportadoras'], 
                     y=Receitas_geradas_por_Transportadora['Receita gerada'], 
                     marker_color='#1f77b4')) 

# Atualizando o layout do gráfico
fig.update_layout(title='Distribuição de Receita por Transportadora', 
                  xaxis_title='Transportadoras', 
                  yaxis_title='Receita Gerada',
                  )  
fig.show()

A análise do desempenho das transportadoras revela que a Transportadora B se destaca como a principal geradora de receita, evidenciando uma liderança no mercado ou uma maior preferência por seus serviços. A Transportadora A, embora apresente um bom desempenho, gera uma receita inferior à da Transportadora B, posicionando-se como uma opção relevante, mas secundária. Já a Transportadora C aparece com a menor receita, sugerindo uma menor participação no mercado ou uma preferência reduzida por parte dos clientes.

In [231]:
# Agrupando os dados por Localização dos clientes e somando as receitas geradas
Receitas_geradas_por_Estado = TabcadeiaSup.groupby('Localização')['Receita gerada'].sum().reset_index()

Receitas_geradas_por_Estado = Receitas_geradas_por_Estado.sort_values(by='Receita gerada', ascending=False)

# Criando o gráfico de barras horizontal
Receitas_geradas_por_Estado_chart = px.bar(Receitas_geradas_por_Estado, 
                   x='Receita gerada', 
                   y='Localização', 
                   orientation='h', 
                   title='Distribuição de Receita por Estado', 
                   color='Localização',
                   color_discrete_sequence=['#1f77b4'])

Receitas_geradas_por_Estado_chart.update_layout(xaxis_title='Receita Gerada', 
                        yaxis_title='Localização',
                        showlegend=False)
Receitas_geradas_por_Estado_chart.show()

São Paulo e Rio de Janeiro se destacam como os estados com as maiores receitas, sugerindo que são os principais mercados consumidores da empresa. Em contraste, Salvador e Belo Horizonte apresentam receitas intermediárias, refletindo um mercado relevante, mas menos expressivo que os líderes. Brasília, por sua vez, registra a menor receita, o que pode estar relacionado a fatores como menor densidade populacional ou menor relevância econômica para a empresa.

**3.1 Análise de SKUs**

In [232]:
# Agrupando os dados por SKU e somando as receitas geradas 
Receitas_geradas_por_SKU = TabcadeiaSup.groupby('Código do produto (SKU)')['Receita gerada'].sum().reset_index()

# Identificando o valor máximo e mínimo
max_value = Receitas_geradas_por_SKU['Receita gerada'].max()
min_value = Receitas_geradas_por_SKU['Receita gerada'].min()

Receitas_geradas_por_SKU['Cor'] = Receitas_geradas_por_SKU['Receita gerada'].apply(lambda x: 'darkblue' if x == max_value else ('red' if x == min_value else '#1f77b4'))

# Criando o gráfico de barras
Receitas_geradas_por_SKU_chart = px.bar(Receitas_geradas_por_SKU, 
                        x='Código do produto (SKU)', 
                        y='Receita gerada', 
                        title='Distribuição de Receita por SKU - Destaque para Máximo e Mínimo',
                        color='Cor',
                        color_discrete_map={'darkblue':'darkblue', '#1f77b4':'#1f77b4', 'red':'red'})

Receitas_geradas_por_SKU_chart.update_layout(showlegend=False)

Receitas_geradas_por_SKU_chart.show()

O gráfico de barras exibe a distribuição da receita gerada pelos SKUs, com a altura das barras representando o valor total de receita de cada SKU.

A análise revela uma grande variabilidade na receita gerada, com alguns produtos se destacando significativamente em relação a outros. O SKU 51 é o líder em vendas, gerando a maior receita entre os SKUs analisados. Em contraste, o SKU 59 apresenta a menor receita.

In [233]:
# Agrupando os dados por SKU e somando os níveis de estoque 
Níveis_de_estoque_por_SKU = TabcadeiaSup.groupby('Código do produto (SKU)')['Níveis de estoque'].sum().reset_index()

# Identificando o valor máximo e mínimo
max_value = Níveis_de_estoque_por_SKU['Níveis de estoque'].max()
min_value = Níveis_de_estoque_por_SKU['Níveis de estoque'].min()

Níveis_de_estoque_por_SKU['Cor'] = Níveis_de_estoque_por_SKU['Níveis de estoque'].apply(lambda x: 'darkblue' if x == max_value else ('red' if x == min_value else '#1f77b4'))

# Criando o gráfico de barras
Níveis_de_estoque_chart = px.bar(Níveis_de_estoque_por_SKU, 
                        x='Código do produto (SKU)', 
                        y='Níveis de estoque', 
                        title='Nível de estoque por SKU - Destaque para Máximo e Mínimo',
                        color='Cor',
                        color_discrete_map={'darkblue':'darkblue', '#1f77b4':'#1f77b4', 'red':'red'})

Níveis_de_estoque_chart.update_layout(showlegend=False)

Níveis_de_estoque_chart.show()

O gráfico de barras ilustra os níveis de estoque para cada SKU, com a altura das barras indicando o total de estoque disponível para cada produto.

A análise mostra uma considerável variação nos níveis de estoque entre os SKUs. Destacam-se os SKUs 12, 51 e 59, que apresentam os maiores níveis de estoque entre todos os analisados. Em contraste, o SKU 68 não está representado no gráfico, indicando que possui um nível de estoque de zero, ou seja, não há valores registrados para esse SKU.

In [234]:
# Agrupando os dados por SKU e somando as Qtd. de produto 
Qtde_de_pedido_por_SKU = TabcadeiaSup.groupby('Código do produto (SKU)')['Quantidades de pedido'].sum().reset_index()

# Identificando o valor máximo e mínimo
max_value = Qtde_de_pedido_por_SKU['Quantidades de pedido'].max()
min_value = Qtde_de_pedido_por_SKU['Quantidades de pedido'].min()

Qtde_de_pedido_por_SKU['Cor'] = Qtde_de_pedido_por_SKU['Quantidades de pedido'].apply(lambda x: 'darkblue' if x == max_value else ('red' if x == min_value else '#1f77b4'))

# Criando o gráfico de barras
Qtde_de_pedido_chart = px.bar(Qtde_de_pedido_por_SKU, 
                        x='Código do produto (SKU)', 
                        y='Quantidades de pedido', 
                        title='Quantidade de pedido por SKU - Destaque para Máximo e Mínimo',
                        color='Cor',
                        color_discrete_map={'darkblue':'darkblue', '#1f77b4':'#1f77b4', 'red':'red'})

Qtde_de_pedido_chart.update_layout(showlegend=False)

Qtde_de_pedido_chart.show()

O gráfico de barras mostra a quantidade de pedidos para cada SKU, com a altura das barras refletindo o total de pedidos registrados para cada produto.

A análise destaca uma ampla variação nas quantidades de pedidos. Os SKUs 0 e 90 se destacam com os maiores volumes de pedidos, liderando a lista entre todos os SKUs analisados. Por outro lado, o SKU 74 apresenta o menor volume de pedidos, evidenciando uma menor demanda em comparação com os demais produtos.

**3.2 Análise de Custos**

In [235]:
# Agrupando os dados por transportadora e somando os custos de envio
Custos_de_envio_por_transportadora = TabcadeiaSup.groupby('Transportadoras')['Custos de envio'].sum().reset_index()

# Criando o gráfico de barras para os custos de envio por transportadora
Custos_por_envio_chart = px.bar(Custos_de_envio_por_transportadora, 
                             x='Transportadoras', 
                             y='Custos de envio', 
                             title='Distribuição dos Custos de Envio por Transportadora',                         
                             color_discrete_sequence=['#1f77b4']) 

# Ajustando o layout para melhor visualização
Custos_por_envio_chart.update_layout(xaxis_title='Transportadoras', 
                                  yaxis_title='Custos de Envio',
                                  )  
                                  
Custos_por_envio_chart.show()

O gráfico revela uma variação significativa nos custos de envio entre as três transportadoras analisadas. A Transportadora B destaca-se por ter o maior custo, indicando que seus serviços são os mais caros. Em contraste, as Transportadoras A e C apresentam custos de envio semelhantes, ambos inferiores ao da Transportadora B. Essas diferenças sugerem variações importantes nos serviços e nas tarifas oferecidas por cada transportadora.

In [236]:
# Agrupando os dados por Modos de Transporte e somando os custos
Custos_por_Modo_de_Transporte = TabcadeiaSup.groupby('Modos de transporte')['Custos'].sum().reset_index()

Custos_por_Modo_de_Transporte = Custos_por_Modo_de_Transporte.sort_values(by='Custos', ascending=False)

# Criando o gráfico de barras horizontal
Custos_chart = px.bar(Custos_por_Modo_de_Transporte, 
                              x='Custos', 
                              y='Modos de transporte', 
                              orientation='h', 
                              title='Distribuição dos Custos por Modo de Transporte',
                              color='Modos de transporte',
                              color_discrete_sequence=['#1f77b4'])

Custos_chart.update_layout(xaxis_title='Custos', 
                                   yaxis_title='Modos de transporte',
                                   showlegend=False)

Custos_chart.show()

O gráfico mostra que o transporte rodoviário é o mais caro entre os modos de transporte analisados, sendo o mais oneroso para a empresa. Em contraste, o transporte marítimo é o mais econômico, oferecendo o menor custo. Os transportes ferroviário e aéreo possuem custos intermediários, com o ferroviário sendo ligeiramente mais caro que o aéreo.

**3.3 Análise da Taxa de Defeitos**

In [237]:
# Agrupando os dados por Tipo de Produto e calculando a média das Taxas de Defeitos
Taxas_de_defeito_por_Tipo_de_Produto = TabcadeiaSup.groupby('Tipo de produto')['Taxas de defeitos'].mean().reset_index()


pie_chart = px.pie(Taxas_de_defeito_por_Tipo_de_Produto, 
                   values='Taxas de defeitos', 
                   names='Tipo de produto', 
                   title='Taxas Médias de Defeitos por Tipo de Produto', 
                   hole=0.5, 
                   )

pie_chart.update_traces(textposition='outside', textinfo='percent+label', 
                        marker=dict(line=dict(color='#000000', width=1)))

A maior parte dos defeitos é encontrada nos produtos para cuidados com o cabelo (36,9%), seguidos pelos produtos para cuidados com a pele (34,7%) e pelos cosméticos (28,5%). Esse padrão indica um desempenho desigual em termos de qualidade, com os produtos para cabelo apresentando a maior taxa de defeitos e os cosméticos a menor.

In [238]:
# Agrupando os dados por Modo de transporte e calculando a média das Taxas de Defeitos

Taxas_de_defeito_por_Modo_de_Transporte = TabcadeiaSup.groupby('Modos de transporte')['Taxas de defeitos'].mean().reset_index()


pie_chart = px.pie(Taxas_de_defeito_por_Modo_de_Transporte, 
                   values='Taxas de defeitos', 
                   names='Modos de transporte', 
                   title='Taxas Médias de Defeitos por Modo de Transporte', 
                   hole=0.5)

pie_chart.update_traces(textposition='outside', textinfo='percent+label', 
                        marker=dict(line=dict(color='#000000', width=1)))

O transporte rodoviário apresenta a maior taxa de defeitos, de 28,9%, indicando possíveis pontos de falha ou maior exposição a riscos. Os transportes ferroviário e marítimo têm taxas similares de 25,5%, sugerindo também possíveis pontos de falha. Em contraste, o transporte aéreo apresenta a menor taxa de defeitos, de 20,1%, indicando processos possivelmente mais robustos ou menos suscetíveis a problemas.

**4. VISÃO PRELIMINAR DAS CORRELAÇÕES ENTRE VARÍAVEIS-CHAVE NA CADEIA DE SUPRIMENTOS**

In [239]:
# Codificar variáveis categóricas usando one-hot encoding
TabcadeiaSup_encoded = pd.get_dummies(TabcadeiaSup, drop_first=True)

# Selecionar apenas colunas numéricas após a codificação
numeric_columns_encoded = TabcadeiaSup_encoded.select_dtypes(include=[np.number])
correlation_matrix_encoded = numeric_columns_encoded.corr()

print("\nMatriz de Correlação com Variáveis Categóricas Codificadas:")
print(correlation_matrix_encoded)


Matriz de Correlação com Variáveis Categóricas Codificadas:
                            Preço  Disponibilidade  N° de produtos vendidos  \
Preço                    1.000000         0.019083                 0.005740   
Disponibilidade          0.019083         1.000000                 0.087496   
N° de produtos vendidos  0.005740         0.087496                 1.000000   
Receita gerada           0.038424        -0.075170                -0.001641   
Níveis de estoque        0.078261        -0.025900                 0.022189   
Prazos de entrega        0.044855         0.170439                -0.046419   
Quantidades de pedido    0.095819         0.143769                 0.015992   
Tempos de envio          0.071942        -0.051377                 0.087315   
Custos de envio          0.058544        -0.044181                 0.044285   
Prazo de entrega         0.152185        -0.156669                 0.041230   
Volumes de produção     -0.124576         0.050134                 0.1

A correlação entre Preço e Receita gerada é muito baixa (0.038424), indicando que o preço tem pouco impacto na receita gerada. A correlação entre Custos de fabricação e Preço é levemente negativa (-0.184123), sugerindo que um aumento no preço pode estar associado a uma leve redução nos custos de fabricação. A correlação entre Prazos de entrega e Níveis de estoque é muito baixa (0.072571), indicando que variações nos prazos de entrega têm pouco efeito sobre os níveis de estoque. A correlação entre Taxas de defeitos e Custos de fabricação é levemente negativa (-0.147247), sugerindo que maiores taxas de defeitos podem estar associadas a uma pequena redução nos custos de fabricação.

**5. RECOMENDAÇÕES**

**5.1 Sensibilidade ao Preço e Receita**

**Segmentação de Preços:** Para produtos de "Cuidados com os cabelos", considere uma estratégia de preços mais competitiva ou promoções para aumentar a demanda. Para "Cuidados com a pele", explore a possibilidade de implementar preços premium ou estratégias de valor agregado.

**Ajustes de Preço Dinâmico:** Utilize um modelo de precificação dinâmica que ajusta os preços com base na elasticidade da demanda identificada para cada categoria. Isso pode maximizar a receita sem perder clientes sensíveis ao preço.

**💡 Ação:** Ofereça pacotes promocionais para "Cuidados com os cabelos" e crie campanhas de marketing que destacam os benefícios e a exclusividade dos produtos de "Cuidados com a pele"

**5.2 Participação de Mercado por Categoria**

**Foco em Expansão:** Invista em marketing e desenvolvimento de produto para categorias com maior participação de mercado. Para "Cosméticos", que tem uma participação menor, considere campanhas de marketing direcionadas ou lançamentos de novos produtos para aumentar a participação no mercado.

**💡 Ação:** Lançar novas linhas de produtos ou edições limitadas para "Cosméticos" para atrair mais clientes e aumentar a participação no mercado.

**5.3 Segmentação por Gênero**

**Entendimento do Segmento "Desconhecido":** Realize uma análise mais profunda para entender o perfil dos clientes com gênero "Desconhecido" e desenvolver estratégias personalizadas para este segmento.

**Marketing Segmentado:** Crie campanhas direcionadas para os grupos "Feminino" e "Masculino", ajustando as ofertas com base em suas preferências e padrões de compra.

**💡 Ação:** Implementar um sistema de coleta de dados mais eficiente para identificar e categorizar melhor os clientes, melhorando a personalização das campanhas de marketing.

**5.4 Desempenho das Transportadoras**

**Negociações e Parcerias:** Reavalie as condições e contratos com as transportadoras, considerando negociar melhores taxas com a Transportadora B ou explorar parcerias estratégicas.

**Avaliação de Desempenho:** Realize uma análise detalhada dos serviços oferecidos por cada transportadora para identificar oportunidades de melhoria ou otimização.

**💡 Ação:** Implementar um sistema de feedback de clientes para avaliar a satisfação com as transportadoras e ajustar os contratos com base nas avaliações.

**5.5 Análise Regional**

**Estratégia Regional:** Desenvolva estratégias de marketing e expansão para mercados com alta receita e explore novas oportunidades em regiões com menor desempenho, como Brasília.

**Análise de Mercado Local:** Realize estudos de mercado para entender as necessidades e preferências dos consumidores em cada região.

**💡 Ação:** Expandir a presença de mercado em Brasília através de campanhas publicitárias localizadas e promoções especiais.

**5.6 Análise de SKU**

**Gestão de Inventário:** Avalie o desempenho de cada SKU e ajuste os níveis de estoque e estratégias de promoção com base na receita gerada.

**Otimização de Produtos:** Considere descontinuar SKUs com baixo desempenho e focar em produtos que geram maior receita.

**💡 Ação:** Oferecer descontos ou promoções para SKUs com baixo desempenho para aumentar a demanda ou reavaliar a estratégia de marketing para esses produtos.

**5.7 Custos de Envio e Modos de Transporte**

**Otimização de Custos:** Explore opções para reduzir os custos de transporte rodoviário e avaliar se mudanças no modal de transporte podem reduzir os custos totais.

**Negociações de Tarifas:** Negocie tarifas mais competitivas com as transportadoras, especialmente para transportes rodoviários.

**💡 Ação:** Implementar um sistema de otimização de rotas para reduzir os custos de transporte e avaliar a viabilidade de substituição de transporte rodoviário por marítimo ou ferroviário onde aplicável.

**5.8 Análise de Taxas de Defeitos**

**Controle de Qualidade:** Reforce os processos de controle de qualidade para reduzir as taxas de defeitos, especialmente para produtos com maiores taxas.

**Avaliação de Processos Logísticos:** Melhore os processos de embalagem e transporte para reduzir defeitos, especialmente para modais com altas taxas de defeitos.

**💡 Ação:** Implementar um programa de controle de qualidade mais rigoroso para produtos de "Cuidados com os cabelos" e revisar os processos de embalagem para reduzir defeitos.

**Entre em Contato Comigo**

Caso tenha alguma dúvida, sugestão ou apenas queira trocar uma ideia sobre este projeto, não hesite em entrar em contato comigo!

**karldarwin248@gmail.com**

**https://www.linkedin.com/in/karldarwin**